lib/user.rb

```diff
@@ -0,0 +1,22 @@
+class User
+  attr_reader :name, :jokes, :told
+  def initialize(name)
+    @name = name
+    @jokes = []
+    @told = false
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def tell(user, joke)
+    @user = user
+    @user.learn(joke)
+    @told = true
```

Comment:
knowing that you told someone else a joke with this flag was also not something we were explicitly looking for, but my comments later will clarify this for you